### ISSUE TITLE
Initializing service via get_instances_from_dict() not working as expected

### ISSUE THREAD

**Issue by sakalosj** [packit/ogr#383]
Shoudn't be this working?

config:
```
authentication:
  git.stg.centos.org:
    instance_url: https://git.stg.centos.org
    token: ---token---
    type: pagure
```
error:
```
Authentication for url 'https://git.stg.centos.org/source-git/packit-hello-world' is missing in the config.
Traceback (most recent call last):
  File "/home/sakalosj/projects/packit/packit/config/config.py", line 203, in _get_project
    url=url, custom_instances=self.services, **get_project_kwargs
  File "/home/sakalosj/projects/ogr/ogr/factory.py", line 83, in get_project
    kls = get_service_class(url=url, service_mapping_update=service_mapping_update)
  File "/home/sakalosj/projects/ogr/ogr/factory.py", line 144, in get_service_class
    raise OgrException("No matching service was found.")
ogr.exceptions.OgrException: No matching service was found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/sakalosj/projects/packit-service/tmp_local/runner.py", line 63, in <module>
    process_new_pr()
  File "/home/sakalosj/projects/packit-service/tmp_local/runner.py", line 22, in process_new_pr
    SteveJobs().process_message(event=new_pr, source="centosmsg")
  File "/home/sakalosj/projects/packit-service/packit_service/worker/jobs.py", line 320, in process_message
    if project:
  File "/home/sakalosj/projects/packit/packit/config/config.py", line 208, in _get_project
    raise PackitConfigException(msg, ex)
packit.exceptions.PackitConfigException: ("Authentication for url 'https://git.stg.centos.org/source-git/packit-hello-world' is missing in the config.", OgrException('No matching service was found.'))
```

**Comment by lachmanfrantisek** [comment]
Looks like some error on the way to this method:

```python
In [4]: from ogr.factory import get_instances_from_dict  

In [5]: get_instances_from_dict( 
   ...:     { 
   ...:         "git.stg.centos.org": { 
   ...:             "instance_url": "https://git.stg.centos.org", 
   ...:             "token": "token", 
   ...:             "type": "pagure", 
   ...:         } 
   ...:     } 
   ...: )                                                                                                                                                                                                                                      
Out[5]: {<ogr.services.pagure.service.PagureService at 0x7f83d04760d0>}
```

**Comment by lachmanfrantisek** [comment]
This should be fixed by #418
